src/migrate/azext_migrate/_params.py

```diff
@@ -0,0 +1,148 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+# pylint: disable=too-many-lines
+# pylint: disable=too-many-statements
+
+from azure.cli.core.commands.parameters import (
+    tags_type,
+    get_enum_type,
+    resource_group_name_type,
+    get_location_type
+)
+from azext_migrate.actions import (
+    AddMachines
+)
+
+
+def load_arguments(self, _):
+
+    with self.argument_context('migrate location check-name-availability') as c:
+        c.argument('location_name', id_part=None, help='The desired region for the name check.')
+        c.argument('name', id_part=None, help='The name to check for availability')
+
+    with self.argument_context('migrate assessment-options show') as c:
+        c.argument('location_name', id_part=None, help='Azure region in which the project is created.')
+
+    with self.argument_context('migrate projects list') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+
+    with self.argument_context('migrate projects show') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('project_name', id_part=None, help='Name of the Azure Migrate project.')
+
+    with self.argument_context('migrate projects create') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('project_name', id_part=None, help='Name of the Azure Migrate project.')

```

Comment:
project is enough, and it should support both name and resource id

Reply:
Are all the resource references support that kind of feature?